services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/serialization/ResumableFileUploadSerializer.java

```diff
@@ -0,0 +1,180 @@
+     * therefore makes use of the standard JSON marshalling classes.
+     */
+    public static byte[] toJson(ResumableFileUpload upload) {
+        JsonWriter jsonGenerator = JsonWriter.create();
+
+        jsonGenerator.writeStartObject();
+
+        TransferManagerJsonMarshaller.LONG.marshall(upload.fileLength(), jsonGenerator, FILE_LENGTH);
+        TransferManagerJsonMarshaller.INSTANT.marshall(upload.fileLastModified(), jsonGenerator, FILE_LAST_MODIFIED);
+        if (upload.multipartUploadId().isPresent()) {
+            TransferManagerJsonMarshaller.STRING.marshall(upload.multipartUploadId().get(), jsonGenerator,
+                                                          MULTIPART_UPLOAD_ID);
+        }
+        if (upload.partSizeInBytes().isPresent()) {
+            TransferManagerJsonMarshaller.LONG.marshall(upload.partSizeInBytes().getAsLong(),
+                                                        jsonGenerator,
+                                                        PART_SIZE_IN_BYTES);
+        }
+        if (upload.totalNumOfParts().isPresent()) {
+            TransferManagerJsonMarshaller.LONG.marshall(upload.totalNumOfParts().getAsLong(),
+                                                        jsonGenerator,
+                                                        TOTAL_NUM_OF_PARTS);
+        }
+        marshallUploadFileRequest(upload.uploadFileRequest(), jsonGenerator);
+        jsonGenerator.writeEndObject();
+
+        return jsonGenerator.getBytes();
+    }
+
+    /**
+     * At this point we do not need to persist the TransferRequestOverrideConfiguration, because it only contains listeners and
+     * they are not used in the resume operation.
+     */
+    private static void marshallUploadFileRequest(UploadFileRequest fileRequest, JsonWriter jsonGenerator) {
+        jsonGenerator.writeFieldName(UPLOAD_FILE_REQUEST);
+        jsonGenerator.writeStartObject();
+        jsonGenerator.writeFieldName(SOURCE);
+        jsonGenerator.writeValue(fileRequest.source().toString());
+        marshallPutObjectRequest(fileRequest.putObjectRequest(), jsonGenerator);
+        jsonGenerator.writeEndObject();
+    }
+
+    private static void marshallPutObjectRequest(PutObjectRequest putObjectRequest, JsonWriter jsonGenerator) {
+        jsonGenerator.writeFieldName(PUT_OBJECT_REQUEST);
+        jsonGenerator.writeStartObject();
+        validateNoRequestOverrideConfiguration(putObjectRequest);
+        putObjectRequest.sdkFields().forEach(field -> marshallPojoField(field, putObjectRequest, jsonGenerator));
+        jsonGenerator.writeEndObject();
+    }
+
+    private static void validateNoRequestOverrideConfiguration(PutObjectRequest putObjectRequest) {
+        if (putObjectRequest.overrideConfiguration().isPresent()) {
+            log.debug(() -> "ResumableFileUpload PutObjectRequest contains an override configuration that will not be "
```

Comment:
This is also what we decided in pause/resume for download PR https://github.com/aws/aws-sdk-java-v2/pull/3177#discussion_r865223841

My reasoning is that it would likely blow up customers' logging file if we change it to WARN or INFO since this is a per-request log statement